chk/chk.go

```diff
@@ -374,17 +374,31 @@ func GetResponseHasEntries(t testing.TB, getres *spb.GetResponse, wants ...fluen
 
 		switch v := wantProto.Entry.(type) {
 		case *spb.AFTEntry_NextHopGroup:
-			if _, ok := ni.nhg[v.NextHopGroup.GetId()]; !ok {
+			if nhg, ok := ni.nhg[v.NextHopGroup.GetId()]; !ok {
 				t.Fatalf("did not find entry, did not find nexthop group: %s, got:\n%s", v.NextHopGroup, getres)
+			} else {
+				programmedStatusMatch(t, nhg, wantProto)
 			}
 		case *spb.AFTEntry_NextHop:
-			if _, ok := ni.nh[v.NextHop.GetIndex()]; !ok {
+			if nh, ok := ni.nh[v.NextHop.GetIndex()]; !ok {
 				t.Fatalf("did not find entry, did not find nexthop: %s, got:\n%s", v.NextHop, getres)
+			} else {
+				programmedStatusMatch(t, nh, wantProto)
 			}
 		case *spb.AFTEntry_Ipv4:
-			if _, ok := ni.ipv4[v.Ipv4.GetPrefix()]; !ok {
+			if ipv4, ok := ni.ipv4[v.Ipv4.GetPrefix()]; !ok {
 				t.Fatalf("did not find entry, did not find ipv4: %s, got: %s\n", v.Ipv4, getres)
+			} else {
+				programmedStatusMatch(t, ipv4, wantProto)
```

Comment:
```suggestion
			ipv4, ok := ni.ipv4[v.Ipv4.GetPrefix()]
			if !ok {
				t.Fatalf("did not find entry, did not find ipv4: %s, got: %s\n", v.Ipv4, getres)
			} 
			programmedStatusMatch(t, ipv4, wantProto)
```

Reply:
Suggest this layout here and above to avoid the indentation of the other case -- since the `!ok` case will exit via `Fatalf` we don't really need the `else`.